### PR TITLE
Migrate `filetypes` to `file/types`

### PR DESCRIPTION
**Pull request by sergkudinov**

### CODE REVIEW

File: packages/file/src/types/hfile.coffee.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        'backup':
+          $ref: 'module://@nikitajs/file/src/index#/properties/backup'
+        'backup_mode':
+          $ref: 'module://@nikitajs/file/src/index#/properties/backup_mode'
+        'eof':
+          $ref: 'module://@nikitajs/file/src/index#/properties/eof'
+        'encoding':
+          $ref: 'module://@nikitajs/file/src/index#/properties/encoding'
+        'uid':
+          $ref: 'module://@nikitajs/file/src/index#/properties/uid'
```
Comment (wdavidw) [inline packages/file/src/types/hfile.coffee.md:48]: Missing `gid` as well as `mode`. We spoke about a lot of other properties including `local`, `unlink`...

File: packages/file/src/types/krb5_conf.coffee.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          Read the target if it exists and merge its content.
+          """
+        'target':
+          type: 'string'
+          description: """
+          Destination file.
+          """
+      required: ['content']
+
+## Example registering a new realm
```
Comment (wdavidw) [inline packages/file/src/types/krb5_conf.coffee.md:48]: Please follow the guideline shared everywhere across the source code, schema comes after examples.

File: packages/file/src/types/krb5_conf.coffee.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    ? 'Configuration was updated'
+    : 'No change occured' )
+})
+```
+
+## Handler
+
+    handler = ({config}) ->
+      # log message: "Entering file.types.krb5_conf", level: 'DEBUG', module: 'nikita/file/lib/types/krb5_conf'
+      config.target ?= '/etc/krb5.conf'
```
Comment (wdavidw) [inline packages/file/src/types/krb5_conf.coffee.md:74]: json schema default value

**Review by wdavidw — COMMENTED** [review]
Minor comments, overall it looks very good.

**Comment by sergkudinov** [comment]
All the comments resolved, check pls

File: packages/file/src/types/systemd/timesyncd.coffee.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+          """
+        'target':
+          type: 'string', default: '/etc/systemd/timesyncd.conf'
+          description: """
+          File to write.
+          """
+
+This action uses `file.ini` internally, therefore it honors all
+arguments it provides. `backup` is true by default and `separator` is
+overridden by "=".
```
Comment (sergkudinov) [inline packages/file/src/types/systemd/timesyncd.coffee.md:59]: `backup` property in `@file.ini` is one of the type: `string` or `function`. So, we cannot pass `true` as a value. What value can we pass instead?

File: packages/file/src/types/systemd/resolved.coffee.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+          """
+        'target':
+          type: 'string', default: '/usr/lib/systemd/resolved.conf.d/resolved.conf'
+          description: """
+          File to write.
+          """
+
+This action uses `file.ini` internally, therefore it honors all
+arguments it provides. `backup` is true by default and `separator` is
+overridden by "=".
```
Comment (sergkudinov) [inline packages/file/src/types/systemd/resolved.coffee.md:58]: `backup` property in `@file.ini` is one of the type: `string` or `function`. So, we cannot pass `true` as a value. What value can we pass instead?

File: packages/file/src/types/ceph_conf.coffee.md [hunk 6]
```diff
@@ -44,13 +44,13 @@ file][ceph-conf] as Ceph daemons expect it.
...
+          File to write.
           """
       required: ['target']
 
 ## Handler
 
-    handler = ({config, operations: {log}}) ->
+    handler = ({config}) ->
       # log message: "Entering file.types.ceph_conf", level: 'DEBUG', module: 'nikita/lib/file/types/ceph_conf'
       config.target = "#{path.join config.rootdir}" if config.rootdir
```
Comment (wdavidw) [inline packages/file/src/types/ceph_conf.coffee.md:55]: Could you please fix this one, it shall be `#{path.join config.rootdir, config.target}"

File: packages/file/src/types/locale_gen.coffee.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+          """
+      required: ['locales']
+
+## Handler
+
+    handler = ({config}) ->
+      # log message: "Entering file.types.local_gen", level: 'DEBUG', module: 'nikita/lib/file/types/local_gen'
+      config.target = "#{path.join config.rootdir, config.target}" if config.rootdir
+      # Write configuration
+      {data} = await @fs.base.readFile ssh: config.ssh, target: config.target, encoding: 'ascii'
```
Comment (wdavidw) [inline packages/file/src/types/locale_gen.coffee.md:53]: multiple lines, remove ssh property

File: packages/file/src/types/locale_gen.coffee.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+          if match[1] in config.locales
+            locales[i] = match[1]+match[2]
+            status = true
+        if match = /^([\w_\-\.]+)($| .+$)/.exec locale
+          if match[1] not in config.locales
+            locales[i] = '#'+match[1]+match[2]
+            status = true
+      if status
+        data = locales.join '\n'
+        res = await @fs.base.writeFile ssh: config.ssh, target: config.target, content: data
```
Comment (wdavidw) [inline packages/file/src/types/locale_gen.coffee.md:67]: same, multiple lines, remove ssh property

File: packages/file/src/types/my_cnf.coffee.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+## Exports
+
+    module.exports =
+      handler: handler
+      schema: schema
+
+## Dependencies
+
+    utils = require '../utils'
```
Comment (wdavidw) [inline packages/file/src/types/my_cnf.coffee.md:61]: new line at the end of the file, atom has a plugin for it (or by default)

**Review by wdavidw — APPROVED** [review]

**Comment by wdavidw** [comment]
Please apply the lastest change and stash it all into one commit before merging to engine

File: packages/file/src/types/my_cnf.coffee.md (hunk 9 above)
Comment (sergkudinov) [reply packages/file/src/types/my_cnf.coffee.md:61]: > new line at the end of the file, atom has a plugin for it (or by default)

All my commited files have a new line at the end.

File: packages/file/src/types/locale_gen.coffee.md (hunk 7 above)
Comment (sergkudinov) [reply packages/file/src/types/locale_gen.coffee.md]: done

File: packages/file/src/types/locale_gen.coffee.md (hunk 8 above)
Comment (sergkudinov) [reply packages/file/src/types/locale_gen.coffee.md]: done

**Comment by sergkudinov** [comment]
> Please apply the lastest change and stash it all into one commit before merging to engine

done

**Pull request merged**